### ISSUE TITLE
getattr(myclass,key) not working if key changes

### ISSUE THREAD

**Issue by simplynail** [ponyorm/pony#223]
I'm doing a 'DataWarehouse' from multiple MS Excel files. From each workbook I get separate "tables_data" that I then add to the database to proper tables.
**I get error when referring to dynamically assigned attribute_name via getattr, it looks like it uses old/previous value of attribute_name**

Here's models declaration:
```
class MyClass1(db.Entity):
    _table_ = 'table_name1'

    id = orm.PrimaryKey(int, auto=True)
    name = orm.Required(str, nullable=True)
    other_attr = orm.Required(str, nullable=True)

class MyClass2(db.Entity):
    _table_ = 'table_name2'

    id = orm.PrimaryKey(int, auto=True)
    number = orm.Required(str, nullable=True)
    other_attr = orm.Required(str, nullable=True)

class MyClass3(db.Entity):
    _table_ = 'table_name3'

    id = orm.PrimaryKey(int, auto=True)
    number = orm.Required(str, nullable=True)
    other_attr = orm.Required(str, nullable=True)
```

and app code is:
```
detail_classes = [myClass1, myClass2, myClass3]
tables_data = {'table_name1':
                        [{'name':'xy','other_attr':'some_value1'},
                         {'name':'yz','other_attr':'some_value2'}],
                   'table_name2':
                        [{'number':'ab','other_attr':'some_value3'},
                         {'number':'bc','other_attr':'some_value4'}],
                   'table_name3':
                        [{'number':'hi','other_attr':'some_value5'},
                         {'number':'ij','other_attr':'some_value6'}]}

for cls_template in detail_classes:
            # select data for current class template
            data = tables_data[cls_template._table_] 
            # set column name that is unique for current dict
            if cls_template._table_ != 'table_name1':
                key = 'name'
            else:
                key = 'number'
```
**so when I do:**
`selected = list(cls_template.select(lambda c: getattr(c, key) == row[key]))`

I get:
**`AttributeError: Entity MyClass2 does not have attribute name: getattr(c, key)`**

Even though current key == 'number'

**Comment by andreymal** [comment]
Same problem for me

**Comment by kozlovsky** [comment]
I cannot reproduce this bug. The tests I tried works without any problem. On the other side, the code presented contains mistakes. For example, the last `if` condition looks wrong: it should use `==` instead of `!=`.

I suspect, the problem was in application logic and not in Pony implementation of `getattr`.

**Comment by andreymal** [comment]
I'll try to write test code later.

My code has no logic errors; if I replace `lambda x: ...` to `eval('lambda x: ...')`, then all works good (but memory leaks of course :)

**Comment by andreymal** [comment]
It's really hard to reproduce it >_<

I don't know if this is case of author of this issue, but this is my case:

```python
from pony.orm import *

db = Database()


class Test1(db.Entity):
    foo = Required(str, 4)
    bar = Required(str, 4)
    PrimaryKey(foo, bar)  # optional: can be removed


class Test2(db.Entity):
    pass


class Test3(db.Entity):
    foo = Required(str, 4)
    bar = Required(str, 4)
    PrimaryKey(foo, bar)  # required


db.bind('sqlite', ':memory:')
db.generate_mapping(create_tables=True)


with db_session:
    for entity, attr in [
        (Test1, 'foo'),
        (Test2, 'id'),
        (Test3, 'foo'),
    ]:
        print('Entity: {}, attr: {}, exists: {}'.format(
            entity, attr, hasattr(entity, attr)
        ))
        entity.select(lambda x: getattr(x, attr) == 'foo')

```

```
Entity: <class '__main__.Test1'>, attr: foo, exists: True
Entity: <class '__main__.Test2'>, attr: id, exists: True
Entity: <class '__main__.Test3'>, attr: foo, exists: True
Traceback (most recent call last):
AttributeError: Entity Test3 does not have attribute id: getattr(x, attr)
```

**Comment by kozlovsky** [comment]
Fixed in d1ca6775!
It was a tricky issue, thank you @simplynail and @andreymal for the test cases!
